### ISSUE TITLE
[Performance] Mention cachetool for clearing opcode caches?

### ISSUE THREAD

**Issue by apfelbox** [symfony/symfony-docs#7793]
In the performance article, the section about byte code caches mentions the problem with clearing the APC/OpCache from the CLI.

The article mentions a working, but rather impractical way:

> You either need to restart the web server or call the apc_clear_cache() or opcache_reset() functions via the web server (i.e. by having these in a script that you execute over the web).

If you are running PHP via FCGI (like FPM) you can use the [cachetool](https://github.com/gordalina/cachetool) to send the cache clear command directly via the FCGI protocol. That prevents exposing a publicly callable cache-clear script (which might be a DDoS attack vector).

So my question: would be beneficial to mention this tool here as a better solution to this problem?

**Comment by xabbuh** [comment]
I haven't seen the tool before. Did you use it? If so, what's your experience with it?

**Comment by apfelbox** [comment]
I have used it and it works reliably and is very easy to use.

It sends request directly via FCGI, so that you can run several PHP commands in the web server PHP pool.

Practically all our dev & production deployment scripts contain now:

```bash
FCGI_PORT=9100
cachetool --fcgi="127.0.0.1:$FCGI_PORT" stat:clear
echo "Stat cache cleared."
cachetool --fcgi="127.0.0.1:$FCGI_PORT" opcache:reset
echo "Opcache cleared."
cachetool --fcgi="127.0.0.1:$FCGI_PORT" opcache:status
echo "done."
```

It is a big step up from putting a publicly reachable PHP script in your app that is called via cURL on deployment.

**Comment by xabbuh** [comment]
Looks like it could make sense to document this tool. What do the other @symfony/team-symfony-docs members think?

**Comment by javiereguiluz** [comment]
👍  clearing the OPCache is a must and the current solutions (graceful web server restart or fake page loading calling `opcache_reset()`) are not entirely satisfying. The mentioned tool looks popular and mature enough to me.

**Comment by HeahDude** [comment]
Great, let's do this! @apfelbox will you open a PR? Thanks!

**Comment by scuben** [comment]
I think beside giving the community the information about this tool and how to clear the opcache nicely we could also spread the information about atomic deployments. @jwage did some nice findings on this with nginx using `$realpath_root` without the deprecated `AppClassLoader` (see https://twitter.com/jwage/status/930862249648025600 and https://twitter.com/jwage/status/930992190947504128).

**Comment by javiereguiluz** [comment]
Fixed by #9107.
